File: Stegnography/decode.py
import cv2
import os
import tkinter as tk
from tkinter import filedialog

def select_image():
    root = tk.Tk()
    root.withdraw()
    file_path = filedialog.askopenfilename(title="Select the Encrypted Image", filetypes=[("Image Files", "*.png")])
    return file_path
image_path = select_image()
if not image_path:
    print("No image selected. Exiting...")
    exit()
img = cv2.imread(image_path)
if img is None:
    print("Error: Unable to load the image.")
    exit()
password_file = os.path.join(os.path.dirname(image_path), "password.txt")
if not os.path.exists(password_file):
    print("Error: Password file not found!")
    exit()

with open(password_file, "r") as f:
    stored_password = f.read().strip()
pas = input("Enter passcode for decryption: ")

if pas != stored_password:
    print("ERROR: Incorrect passcode!")
    exit()
msg_length = int(img[0, 0, 0]) + (int(img[0, 0, 1]) * 256)
message_bytes = []
index = 0
for row in range(img.shape[0]):
    for col in range(img.shape[1]):
        for color in range(3):
            if row == 0 and col == 0:
                continue 
            if index < msg_length:
                message_bytes.append(int(img[row, col, color])) 
                index += 1
            else:
                break
        if index >= msg_length:
            break
    if index >= msg_length:
        break
message = bytes(message_bytes).decode('utf-8', errors='ignore')
print("Decrypted message:", message)

File: decode.java
import javax.imageio.ImageIO;
import javax.swing.*;
import java.awt.image.BufferedImage;
import java.awt.*;
import java.io.*;

public class decode {
    public static void main(String[] args) throws Exception {
        JFileChooser chooser = new JFileChooser();
        chooser.setDialogTitle("Select the Encrypted Image");
        int result = chooser.showOpenDialog(null);

        if (result != JFileChooser.APPROVE_OPTION) {
            System.out.println("No image selected. Exiting...");
            return;
        }

        File imageFile = chooser.getSelectedFile();
        BufferedImage img = ImageIO.read(imageFile);
        if (img == null) {
            System.out.println("Error: Unable to load the image.");
            return;
        }

        File passwordFile = new File(imageFile.getParent(), "password.txt");
        if (!passwordFile.exists()) {
            System.out.println("Error: Password file not found!");
            return;
        }

        BufferedReader reader = new BufferedReader(new FileReader(passwordFile));
        String storedPassword = reader.readLine().trim();
        reader.close();

        String inputPassword = JOptionPane.showInputDialog("Enter passcode for decryption:");
        if (!storedPassword.equals(inputPassword)) {
            System.out.println("ERROR: Incorrect passcode!");
            return;
        }

        Color c = new Color(img.getRGB(0, 0));
        int msgLen = c.getRed() + (c.getGreen() * 256);

        byte[] msgBytes = new byte[msgLen];
        int index = 0;
        outer:
        for (int y = 0; y < img.getHeight(); y++) {
            for (int x = 0; x < img.getWidth(); x++) {
                if (x == 0 && y == 0) continue;

                Color pixel = new Color(img.getRGB(x, y));
                if (index < msgLen) msgBytes[index++] = (byte) pixel.getRed();
                if (index < msgLen) msgBytes[index++] = (byte) pixel.getGreen();
                if (index < msgLen) msgBytes[index++] = (byte) pixel.getBlue();
                if (index >= msgLen) break outer;
            }
        }

        String message = new String(msgBytes, "UTF-8");
        System.out.println("Decoded Message: " + message);
    }
}
